### DISCUSSION TITLE
Problem with UI interactive buttons when developing for HoloLens 2 with MRTK3

### DISCUSSION THREAD

**Discussion by Jibbevm** [microsoft/MixedRealityToolkit-Unity#11820] [Q&A] [answered]
Hi everyone,
For my graduation project I'm designing a MR application for the Hololens 2. I've created the application in Unity with the help of MRTK3. All the UI elements are canvas based with scriped buttons to switch between the different UI Slates, all in the same scene. Everything works fine in the Unity editor during runtime, but as soon as I build the application and try to use the application it doesn't work. When I start the application on the HoloLens I just see one of the canvas based UI slates and the buttons have lost their functionality. I'll add the screen recording from my Unity editor during runtime (I uploaded it to YouTube). I hope somebody knows or sees anything that will solve this problem!

Thank you all in advance!

https://youtu.be/k3hRJaPHQ_w

**Comment by dudziakl** [comment, marked as answer]
Check if you have in your project Active Input system set to Input System Package or Both
![image](https://github.com/microsoft/MixedRealityToolkit-Unity/assets/12583808/710d3331-9a70-47a9-9b7f-25c1086f321c)
and if you have on your default scene EventSystem object with two components Event System and XR UI Input Module. Make sure that you don't have two EventSystem objects with different settings.
![image](https://github.com/microsoft/MixedRealityToolkit-Unity/assets/12583808/d44feb6e-ada9-438b-a221-25f6f4e9a233)
If it doesn't solve the problem, then please compare your project with the VanillaUGUIExample from the MRTK3 sample projects.
https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/blob/main/UnityProjects/MRTKDevTemplate/Assets/Scenes/VanillaUGUIExample.unity

**Reply by Jibbevm** [reply]
Thank you for your message! I changed the UI Input Module to the one that you had shown instead of the one Unity adds automatically and now it works!!
